### PR TITLE
CVP resource pool support

### PR DESCRIPTION
**Pull request by ahmadiesa-abu**
Added support for resource pool [ add, delete and update ] and a relation for resource pool to be contained in another resource pool

### CODE REVIEW

File: cloudify_vsphere/resource_pool/__init__.py [hunk 1]
```diff
@@ -22,47 +22,149 @@
...
 @op
 @with_server_client
-def create(ctx, server_client, name, use_external_resource):
-    vmware_resource = server_client._get_obj_by_name(
-        vim.ResourcePool,
-        name,
-    )
-
+def create(ctx, server_client, name, use_external_resource, host_name,
+           cluster_name, pool_spec):
```
Comment (EarthmanT) [inline cloudify_vsphere/resource_pool/__init__.py:96]: Add default values to new parameters, e.g. `host_name=None, cluster_name=None, pool_spec=None`

File: CHANGELOG.txt [hunk 2]
```diff
@@ -1,3 +1,4 @@
+2.18.14: Add Resource Pool Support
```
Comment (EarthmanT) [inline CHANGELOG.txt:1]: New feature, `2.19.0`

File: cloudify_vsphere/resource_pool/__init__.py [hunk 3]
```diff
@@ -22,47 +22,149 @@
...
+
         if not vmware_resource:
             raise NonRecoverableError(
                 'Could not use existing resource_pool "{name}" as no '
                 'resource_pool by that name exists!'.format(
                     name=name,
                 )
             )
+
+        ctx.instance.runtime_properties[RESOURCE_POOL_ID] = vmware_resource.id
```
Comment (EarthmanT) [inline cloudify_vsphere/resource_pool/__init__.py:108]: `use .get()`

File: cloudify_vsphere/resource_pool/__init__.py [hunk 4]
```diff
@@ -22,47 +22,149 @@
...
 
-    ctx.instance.runtime_properties[RESOURCE_POOL_ID] = vmware_resource.id
+        ctx.instance.runtime_properties[RESOURCE_POOL_ID] = \
+            vmware_resource._moId
 
 
 @op
 @with_server_client
-def delete(ctx, name, use_external_resource, **_):
+def delete(ctx, server_client, name, use_external_resource, **_):
```
Comment (EarthmanT) [inline cloudify_vsphere/resource_pool/__init__.py:149]: new params should be added to the end of the fn declaration with default = None.

File: examples/resource-pool.yaml [hunk 5]
```diff
@@ -0,0 +1,63 @@
+tosca_definitions_version: cloudify_dsl_1_3
+
+imports:
+  - https://cloudify.co/spec/cloudify/4.6/types.yaml
```
Comment (EarthmanT) [inline examples/resource-pool.yaml:4]: Let's use `5.1.`

File: plugin.yaml [hunk 6]
```diff
@@ -6,8 +6,8 @@ plugins:
   vsphere:
     executor: central_deployment_agent
     package_name: cloudify-vsphere-plugin
-    package_version: '2.18.13'
-    source: https://github.com/cloudify-cosmo/cloudify-vsphere-plugin/archive/2.18.13.zip
+    package_version: '2.18.14'
```
Comment (EarthmanT) [inline plugin.yaml:9]: `2.19.0`

File: plugin.yaml [hunk 7]
```diff
@@ -587,11 +587,47 @@ node_types:
           Currently, resource pools cannot be created or deleted,
           and this node type exists only for compatibility with the NSX/Content
           Library plugin.
-        default: True
+        default: False
```
Comment (EarthmanT) [inline plugin.yaml:590]: As this is YAML, this should technically be `false` not `False`.

**Review by EarthmanT — CHANGES_REQUESTED** [review]

File: cloudify_vsphere/resource_pool/__init__.py (hunk 3 above)
Comment (ahmadiesa-abu) [reply cloudify_vsphere/resource_pool/__init__.py:108]: 'vim.ResourcePool' object has no attribute 'get'

**Review by EarthmanT — APPROVED** [review]

**Pull request merged**